lib/linalg/Matrix.h

```diff
@@ -946,6 +946,19 @@ class Matrix
     void
     orthogonalize();
 
+    /**
+     * @brief Orthonormalizes the matrix's last column, assuming the previous
+	 * columns are already orthonormal.
+	 * 
+	 * By default, the function considers the whole matrix.
+	 * If input parameter ncols < d_num_cols, then a subset of the matrix
+	 * is considered. 
+	 * This allows one to reorthonormalize the matrix every time a new column
+	 * is added, assuming the previous columns have remained unchanged.
+     */
+	void
+	orthogonalize_last(const int ncols = d_num_cols);
```

Comment:
@dylan-copeland 
In this function, I want to set the default value for the input argument to be the number of columns of the matrix calling this method. In this way, if no input argument is given, the last column is orthonormalized. If a value smaller than `d_num_cols` is given, then the function considers a subset of the full matrix, allowing the user to reorthonormalize every time a new column is added to a matrix with a larger allocated memory size. 

Given that it is a method of class `Matrix`, my understanding is that the function should have access to the `d_num_cols` variable. But when trying to compile I get the following error: "a nonstatic member reference must be relative to a specific object". The same happens if I try to use the `numColumns()` method instead.

Reply:
I don't think a class member (here, `d_num_cols`) can be used as the default value for a function input argument. Instead, you could set to `d_num_cols` inside the function. For example, the default could be something meaningless, like `const int ncols = -1`, which serves as a flag inside the function: `if (ncols < 0) ncols = d_num_cols;`